src/main/java/com/Acrobot/ChestShop/Listeners/ItemInfoListener.java

```diff
@@ -30,93 +30,108 @@
  * @author Acrobot
  */
 public class ItemInfoListener implements Listener {
-
-    @EventHandler
-    public static void addRepairCost(ItemInfoEvent event) {
+    
+    @EventHandler(priority =  EventPriority.LOW, ignoreCancelled = true)
+    public static void messageHandler (ItemInfoEvent event) {
+        CommandSender sender = event.getSender();
+        ItemStack item = event.getItem();
+        iteminfo.send(sender);
+        try {
+            iteminfo_fullname.send(sender, "item", MaterialUtil.getName(item));
+        } catch (IllegalArgumentException e) {
+            event.getSender().sendMessage(ChatColor.RED + "Error while generating full name. Please contact an admin or take a look at the console/log!");
+            ChestShop.getPlugin().getLogger().log(Level.SEVERE, "Error while generating full item name", e);
+            return;
+        }
+        
+        try {
+            iteminfo_shopname.send(sender, "item", MaterialUtil.getSignName(item));
+        } catch (IllegalArgumentException e) {
+            sender.sendMessage(ChatColor.RED + "Error while generating shop sign name. Please contact an admin or take a look at the console/log!");
+            ChestShop.getPlugin().getLogger().log(Level.SEVERE, "Error while generating shop sign item name", e);
+        }
+    }
+    
+    @EventHandler(ignoreCancelled = true)
+    public static void addRepairCost (ItemInfoEvent event) {
```

Comment:
Please follow the general code style of the project (e.g. no space in front of method parenthesis here which results in this being counted as a change when it really isn't)